test/parser/LiteralOrIriTest.cpp

```diff
@@ -21,6 +22,57 @@ TEST(IriTest, IriCreation) {
               asStringViewUnsafe(iri.getContent()));
 }
 
+TEST(IriTest, getBaseIri) {
+  Iri iri1 = Iri::fromIriref("<http://purl.uniprot.org/uniprot/>");
+  EXPECT_THAT("<http://purl.uniprot.org/>",
+              iri1.getBaseIri().toStringRepresentation());
+  Iri iri2 = Iri::fromIriref("<http://purl.uniprot.org/>");
+  EXPECT_THAT("<http://purl.uniprot.org/>",
+              iri2.getBaseIri().toStringRepresentation());
+  Iri iri3 = Iri::fromIriref("<http://purl.uniprot.org>");
+  EXPECT_THAT("<http://purl.uniprot.org>",
+              iri3.getBaseIri().toStringRepresentation());
+  Iri iri4 = Iri::fromIriref("<>");
+  EXPECT_THAT("<>", iri4.getBaseIri().toStringRepresentation());
+}
+
+TEST(IriTest, emptyIri) {
+  EXPECT_TRUE(Iri{}.empty());
+  EXPECT_FALSE(Iri::fromIriref("<http://www.wikidata.org>").empty());
+}
+
+TEST(IriTest, fromIrirefConsiderBase) {
+  // Helper lambda that calls `Iri::fromIrirefConsiderBase` with the two base
+  // IRIs and returns the results as a string (including the angle brackets).
+  Iri baseForRelativeIris;
+  Iri baseForAbsoluteIris;
+  auto fromIrirefConsiderBase = [&baseForRelativeIris, &baseForAbsoluteIris](
+                                    std::string_view iriStringWithBrackets) {
+    return Iri::fromIrirefConsiderBase(iriStringWithBrackets,
+                                       baseForRelativeIris, baseForAbsoluteIris)
+        .toStringRepresentation();
+  };
+
+  // Check that it works for "real" base IRIs.
+  baseForRelativeIris = Iri::fromIriref("<http://.../uniprot/>");
+  baseForAbsoluteIris = Iri::fromIriref("<http://.../>");
+  EXPECT_THAT(fromIrirefConsiderBase("<http://purl.uniprot.org/uniprot/>"),
+              "<http://purl.uniprot.org/uniprot/>");
+  EXPECT_THAT(fromIrirefConsiderBase("<UPI001AF4585D>"),
+              "<http://.../uniprot/UPI001AF4585D>");
+  EXPECT_THAT(fromIrirefConsiderBase("</prosite/PS51927>"),
+              "<http://.../prosite/PS51927>");
```

Comment:
This is also an alternative how to do it.
But these can be a simple `EXPECT_EQ` (EXPECT_THAT also works but is overkill).